src/rust/ensogl/src/math/geometry/triangle.rs

```diff
@@ -0,0 +1,33 @@
+use core::f32::consts::PI;
+use crate::display::shape::Var;
+use crate::math::algebra::{Sin, Cos, Sqrt};
+use crate::math::topology::unit::{Radians, Angle};
+use std::ops::*;
+
+pub struct Triangle<S> {
```

Comment:
I dont like this triangle definition TBH. It allows me to construct impossible triangles. Structs allowing to create impossible data are rather error prone.

Reply:
I've changed the struct so it can not be instantiated outside of this module. Here it is only meant to serve as a return type for the helper functions.